### PR TITLE
HLSL/Shapes

### PR DESCRIPTION
**Pull request by nahiim**
## Description
HLSL/shapes fixes

## TODO list:
NBL_.._ARG_REF 
sized floats, ints and uints

### CODE REVIEW

File: include/nbl/builtin/hlsl/shapes/beziers.hlsl [hunk 1]
```diff
@@ -17,17 +17,17 @@ namespace shapes
     // GH link https://github.com/erich666/GraphicsGems/blob/master/gems/Roots3And4.c
     // Credits to Doublefresh for hinting there
     // returns the roots, and number of filled in real values under numRealValues
-    float2 SolveQuadratic(float3 c, out int numRealValues)
+    float32_t2 SolveQuadratic(float32_t3 c, NBL_REF_ARG(int32_t) numRealValues)
```
Comment (devshgraphicsprogramming) [inline include/nbl/builtin/hlsl/shapes/beziers.hlsl:20]: @Erfan-Ahmadi should he touch this file, and if so should it be on another PR?

File: include/nbl/builtin/hlsl/shapes/aabb.hlsl [hunk 2]
```diff
@@ -17,32 +17,32 @@ namespace shapes
 struct AABB_t
 {
     //
-    void addPoint(const float3 pt)
+    void addPoint(const float32_t3 pt)
```
Comment (devshgraphicsprogramming) [inline include/nbl/builtin/hlsl/shapes/aabb.hlsl:20]: can you template (except for the beziers) the shapes on a `scalar_t` then we can swap the type used to `float16_t` or `int16_t`

if you want a `float32_t3` then you simply do
```cpp
using vec3_t = vector<scalar_t,3>;
```
and use that typedef throughout the class

File: include/nbl/builtin/hlsl/shapes/beziers.hlsl (hunk 1 above)
Comment (Erfan-Ahmadi) [reply include/nbl/builtin/hlsl/shapes/beziers.hlsl:20]: Stay out of "shapes" that's my territory

Comment (devshgraphicsprogramming) [reply include/nbl/builtin/hlsl/shapes/beziers.hlsl:20]: ok @nahiim then I guess you can remove the commit that changes beziers (re-add/cherrypick the commits onto `nahim_shapes2`)

Comment (devshgraphicsprogramming) [reply include/nbl/builtin/hlsl/shapes/beziers.hlsl:20]: > Stay out of "shapes" that's my territory

btw @Erfan-Ahmadi  did you guys template the beziers on a `float_t` yet?
P.S. note the other transformations Nahim did to get rid of HLSL keywords like `out`

Comment (Erfan-Ahmadi) [reply include/nbl/builtin/hlsl/shapes/beziers.hlsl:20]: this is all redundant work, and this file is way outdated, we had done this like two weeks ago
please don't touch shapes.

https://github.com/Devsh-Graphics-Programming/Nabla/blob/cad_shaders_work_hatches/include/nbl/builtin/hlsl/shapes/beziers.hlsl
